rule-engine/rule-engine-components/src/main/java/org/thingsboard/rule/engine/action/TbCreateAlarmNode.java

```diff
@@ -190,6 +191,22 @@ private Alarm getAlarmFromMessage(TbContext ctx, TbMsg msg) throws IOException {
         return Futures.transform(asyncUpdated, a -> new TbAlarmResult(false, true, false, a), MoreExecutors.directExecutor());
     }
 
+    private void updateStatus(Alarm existingAlarm, Alarm msgAlarm) {
+        if (!existingAlarm.getStatus().isCleared() && msgAlarm.getStatus().isCleared()) {
+            throw new RuntimeException("The 'Alarm Create' node cannot clear the alarm!");
+        }
+        if (existingAlarm.getStatus().isAck() && !msgAlarm.getStatus().isAck()) {
+            throw new RuntimeException("An already acknowledged alarm cannot be made unacknowledged!");
+        }
+
+        if (existingAlarm.getStatus().isCleared()) {
+            existingAlarm.setStatus(AlarmStatus.CLEARED_ACK);
+        } else {
+            existingAlarm.setStatus(AlarmStatus.ACTIVE_ACK);
+        }
+        existingAlarm.setAckTs(System.currentTimeMillis());
```

Comment:
What if the status of existing alarm is ACTIVE_UNACK?.. An alarm will just become acknowledged for no reason?